test/Microsoft.TestPlatform.AcceptanceTests/ExecutionTests.cs

```diff
@@ -263,5 +263,79 @@ public void IncompatibleSourcesWarningShouldBeDisplayedInTheConsoleOnlyWhenRunni
                 this.StdOutputContains(expectedWarningContains);
             }
         }
+
+        [TestMethod]
+        [NetFullTargetFrameworkDataSource]
+        public void ExitCodeShouldReturnOneWhenFailWhenNoTestsFoundParameterSetToTrueAndNoTestAvailableWithFilter(RunnerInfo runnerInfo)
+        {
+            SetTestEnvironment(this.testEnvironment, runnerInfo);
+
+            var assemblyPaths = this.BuildMultipleAssemblyPath("SimpleTestProject2.dll").Trim('\"');
+
+            var arguments = PrepareArguments(assemblyPaths, this.GetTestAdapterPath(), string.Empty, this.FrameworkArgValue, runnerInfo.InIsolationValue);
+            
+            // Setting /TestCaseFilter to random test name, which does not exists
+            arguments = string.Concat(arguments, " /TestCaseFilter:SomeRandomTestName");
+
+            // set FailWhenNoTestsFound to true
```

Comment:
this comment just states the same thing as code. Change it to explain why, or remove it.